src/EnergyPlus/IntegratedHeatPump.cc

```diff
@@ -1240,171 +1221,155 @@ void GetIHPInput(EnergyPlusData &state)
         // ErrorsFound will be set to True if problem was found, left untouched otherwise
         VerifyUniqueCoilName(state, CurrentModuleObject, AlphArray(1), ErrorsFound, CurrentModuleObject + " Name");
 
-        state.dataIntegratedHP->IntegratedHeatPumps(DXCoilNum).Name = AlphArray(1);
-        state.dataIntegratedHP->IntegratedHeatPumps(DXCoilNum).IHPtype = "AIRSOURCE_IHP";
+        auto &ihp = state.dataIntegratedHP->IntegratedHeatPumps(CoilCounter);
+
+        ihp.Name = AlphArray(1);
+        ihp.IHPtype = "AIRSOURCE_IHP";
```

Comment:
Should be an `enum` that can be converted to string for display purposes using a `constexpr std::array<std::string_view>`.